### PR TITLE
pkp/pkp-lib#1825 Move HTML view into iframe

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php [hunk 1]
```diff
@@ -103,36 +127,38 @@ function _getHTMLContents($request, $galley) {
 		);
 
 		foreach ($embeddableFiles as $embeddableFile) {
-			$fileUrl = $request->url(null, 'article', 'download', array($galley->getSubmissionId(), $galley->getBestGalleyId($journal), $embeddableFile->getFileId()));
+			$params = array();
+			if ($embeddableFile->getFileType()=='text/plain' || $embeddableFile->getFileType()=='text/css') $params['inline']='true';
+			$fileUrl = $request->url(null, 'article', 'download', array($galley->getSubmissionId(), $galley->getBestGalleyId($journal), $embeddableFile->getFileId()), $params);
```
Comment (bozana) [inline plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php]: Would here be 'important' to consider submission best id?

File: plugins/generic/htmlArticleGalley/display.tpl [hunk 2]
```diff
@@ -35,7 +35,7 @@
 	</header>
 
 	<div id="htmlContainer">
-		<iframe name="htmlFrame" src="{url page="article" op="download" path=$article->getBestArticleId($currentJournal)|to_array:$galley->getBestGalleyId($currentJournal) inline=true}" width="100%" height="100%" style="border: 0px;" allowfullscreen webkitallowfullscreen></iframe>
+		<iframe name="htmlFrame" src="{url page="article" op="download" path=$article->getBestArticleId($currentJournal)|to_array:$galley->getBestGalleyId($currentJournal) inline=true}" allowfullscreen webkitallowfullscreen></iframe>
```
Comment (bozana) [inline plugins/generic/htmlArticleGalley/display.tpl]: The best object id function do not need the journal parameter any more.

File: plugins/generic/htmlArticleGalley/display.tpl [hunk 3]
```diff
@@ -7,7 +7,36 @@
...
+<html lang="{$currentLocale|replace:"_":"-"}" xml:lang="{$currentLocale|replace:"_":"-"}">
+<head>
+	<meta http-equiv="Content-Type" content="text/html; charset={$defaultCharset|escape}" />
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<title>{translate key="article.pageTitle" title=$title}</title>
+
+	{load_header context="frontend" headers=$headers}
+	{load_stylesheet context="frontend" stylesheets=$stylesheets}
+	{load_script context="frontend" scripts=$scripts}
+</head>
```
Comment (bozana) [inline plugins/generic/htmlArticleGalley/display.tpl]: Super minor thing, but could you use `{include file="core:common/headerHead.tpl"}` here?

Comment (asmecher) [inline plugins/generic/htmlArticleGalley/display.tpl]: Done, thanks!

File: plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php (hunk 1 above)
Comment (asmecher) [inline plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php]: Yup, fixed.

**Pull request merged**